localstack-core/localstack/services/apigateway/integration.py

```diff
@@ -718,7 +718,11 @@ def invoke(self, invocation_context: ApiInvocationContext):
 
         action = None
         invoke_args = {"Bucket": bucket, "Key": object_key}
-        match invocation_context.method:
+        integration_method = integration.get("httpMethod") or invocation_context.method
+        if integration_method == "ANY":
+            integration_method = invocation_context.method
+
+        match integration_method:
```

Comment:
```suggestion
        if not (integration_method := integration.get("httpMethod")) or integration_method == "ANY":
            integration_method = invocation_context.method

        match integration_method:
```

Reply:
Thanks I will update according to this comment. I also noticed a missing implementation of requestParameters for integration uri, can I post this fix inside this PR?

```
    def apply_integration_request_parameters(uri: str, integration: Dict[str, Any], invocation_context: ApiInvocationContext):
        groups = re.findall(r"({[a-zA-Z]+})", uri)
        request_parameters = integration.get("requestParameters", {})
        for group in groups:
            key = group.replace("{", "").replace("}", "")
            request_param_key = f"integration.request.path.{key}"
            associatedValueKey = request_parameters.get(request_param_key)
            if associatedValueKey:
                value = invocation_context.context.get(associatedValueKey.replace("context.", ""))
                if (value):
                    uri = uri.replace(group, value)
        return uri

    #...

    uri = apply_integration_request_parameters(uri, integration, invocation_context)


```

Reply:
If that's ok for you you can go ahead and merge :)